packages/core/src/interpreter.ts

```diff
@@ -484,6 +490,12 @@ export class Interpreter<
     };
   }
 
+  public getPersistedState(): PersistedFrom<TBehavior> | undefined {
+    return this.behavior.getPersistedState?.(
+      this._state ?? this._getInitialState()
```

Comment:
Couldn't this be simplified to this?
```suggestion
      this.getSnapshot()
```

Reply:
It cannot because the snapshot is not always equal to the internal state (this is only true for state machines and reducers)

Reply:
So perhaps let's just initialize `this._state` in the `constructor` and use `this._state` here. Without the fallback to the `this._getInitialState()`. Do we get anything from this information being computed lazily?

Reply:
Tried this quickly; not easy to do this (tests fail). Help wanted

Reply:
It's likely "just" a part of [this](https://github.com/statelyai/xstate/pull/3743#discussion_r1122973161). I can play more with this PR here and that proposed patch to understand better how the final implementation could look like